src/create_character/create_character.cpp

```diff
@@ -55,39 +49,17 @@ bool CheckAndOverwriteSaveFile()
  * @param main_stats The stats that have been rolled
  * @param array_length The length of the array of stats still left to be placed
  */
-void SetAbilityScore(MainCharacter& main_character, int ability_score_pos, int main_stats[], int array_length)
+void SetAbilityScore(MainCharacter& main_character, int ability_score_pos, std::vector<int>& main_stats)
 {
 	std::string ability_scores[6] = {"Strength", "Dexterity", "Consitution", "Intelligence", "Wisdom", "Charisma"};
 
-	// Print the remaining scores.
-	std::cout << "Your remaining scores:" << std::endl;
-	for (int i = 0; i < array_length; i++)
-	{
-		std::cout << i + 1 << ") " << main_stats[i] << std::endl;
-	}
-
-	int stat_choice = 0;
-
 	// Ask for user input.
-	utils::Print({"Which stat would you like for your " + ability_scores[ability_score_pos] +
-		"? (1-" + std::to_string(array_length) + ")"});
-	utils::GetUserInput(stat_choice);
-
-	// Check for valid input, and set the appropriate ability score.
-	while (stat_choice < 1 || stat_choice > array_length)
-	{
-		std::cout << "Sorry, that is an invalid number. Please give me an "
-					 "intger\nbetween 1 and "
-				  << array_length << ": ";
-		utils::GetUserInput(stat_choice);
-	}
+	utils::Print({"Which stat would you like for your " +
+		ability_scores[ability_score_pos] + "?"});
+	int stat_choice = utils::GetUserConstrainedChoice(main_stats);
```

Comment:
This is intended. It is confusing (and even error-prone) allowing selection through indexes when one has to choose between numbers.

Reply:
Maybe we should add something to tell them what we expect them to do, because they just had the ability to select by the number on the list the Class and Race

Reply:
I'd say the message that pops up in the case of strict comparison is pretty explanatory. In any case, how the user interacts with the application, what they can input etc is maybe something we should consider outside of the scope of this issue.